android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/BrightnessModule.java

```diff
@@ -71,10 +97,85 @@ public void setSystemBrightnessAsync(final float brightnessValue, final Promise
         );
         promise.resolve(null);
       } else {
-        promise.reject("E_BRIGHTNESS_PERMISSIONS", "WRITE_SETTINGS not granted");
+        promise.reject("ERR_BRIGHTNESS_PERMISSIONS", "WRITE_SETTINGS not granted");
       }
     } catch (Exception e){
-      promise.reject("E_BRIGHTNESS", e.getMessage());
+      promise.reject("ERR_BRIGHTNESS_SYSTEM", "Failed to set the system brightness value", e);
+    }
+  }
+
+  @ReactMethod
+  public void useSystemBrightnessAsync(final Promise promise) {
+    final Activity activity = getCurrentActivity();
+    activity.runOnUiThread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          WindowManager.LayoutParams lp = activity.getWindow().getAttributes();
+          lp.screenBrightness = WindowManager.LayoutParams.BRIGHTNESS_OVERRIDE_NONE;
+          activity.getWindow().setAttributes(lp); // must be done on UI thread
+          promise.resolve(null);
+        } catch (Exception e) {
+          promise.reject("ERR_BRIGHTNESS", "Failed to set the brightness of the current screen", e);
+        }
+      }
+    });
+  }
+
+  @ReactMethod
+  public void getSystemBrightnessModeAsync(Promise promise) {
+    try {
+      int brightnessMode = Settings.System.getInt(
+          getCurrentActivity().getContentResolver(),
+          Settings.System.SCREEN_BRIGHTNESS_MODE
+      );
+      promise.resolve(brightnessModeNativeToJS(brightnessMode));
+    } catch (Exception e) {
+      promise.reject("ERR_BRIGHTNESS_MODE", "Failed to get the system brightness mode", e);
+    }
+  }
+
+  @ReactMethod
+  public void setSystemBrightnessModeAsync(int brightnessMode, Promise promise) {
+    try {
+      // we have to just check this every time
+      // if we try to store a value for this permission, there is no way to know if the user has changed it
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && Settings.System.canWrite(getCurrentActivity()) || Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+        Settings.System.putInt(
+            getCurrentActivity().getContentResolver(),
+            Settings.System.SCREEN_BRIGHTNESS_MODE,
+            brightnessModeJSToNative(brightnessMode)
+        );
+        promise.resolve(null);
+      } else {
+        // TODO: what to do if we are rejecting but there is no native error?
+        // should we create a new Exception in order to get the native stack trace?
```

Comment:
To add to this -- I think this can happen in a separate PR at any time. It's pretty self-encapsulated in the sense that it's just the Promise / bridge that needs to be modified, while also cross-cutting across all modules => I don't think it needs to be grouped with this Brightness PR.